py/obj.c

```diff
@@ -602,3 +602,9 @@ mp_obj_t mp_generic_unary_op(mp_unary_op_t op, mp_obj_t o_in) {
             return MP_OBJ_NULL;      // op not supported
     }
 }
+
+void *_m_new_obj_with_type(size_t num_bytes, const mp_obj_type_t *type) {
+    mp_obj_base_t *base = (mp_obj_base_t *)m_malloc(num_bytes);
```

Comment:
Maybe `m_malloc()` can take the type (or have a new version of `m_malloc()`), then it would eliminate a C call altogether?

Reply:
`m_malloc()` is pretty small, wouldn't cost much to duplicate it to `m_malloc_typed(...)`.

Reply:
There was a very small performance gain for this but it cost almost all of the byte saving.